stdlib/public/Concurrency/Task.cpp

```diff
@@ -208,6 +166,17 @@ static void completeTask(AsyncTask *task, ExecutorRef executor,
     task->completeFuture(context, executor);
   }
 
+  // Offer the future to the parent task group's channel.
```

Comment:
When a future that is a group child is completed, it also must offer itself to the readyQueue of the parent.
Thanks to this the readyQueue can maintain a FIFO (by completion order) queue of the tasks as they complete.

This allows us to implement "first n results" in terms of task groups.